measures/UpgradeCosts/measure.rb

```diff
@@ -0,0 +1,282 @@
+      hpxml.walls.each do |wall|
+        next unless wall.is_exterior
+
+        cost_mult += wall.area
+      end
+    elsif cost_mult_type == 'Wall Area, Below-Grade (ft^2)'
+      hpxml.foundation_walls.each do |foundation_wall|
+        cost_mult += foundation_wall.area
+      end
+    elsif cost_mult_type == 'Floor Area, Conditioned (ft^2)'
+      cost_mult += hpxml.building_construction.conditioned_floor_area
+    elsif cost_mult_type == 'Floor Area, Attic (ft^2)'
+      hpxml.frame_floors.each do |frame_floor|
+        next unless frame_floor.is_thermal_boundary
+        next unless frame_floor.is_interior
+
+        if frame_floor.is_floor
+          next unless [HPXML::LocationAtticVented, HPXML::LocationAtticUnvented].include?(frame_floor.interior_adjacent_to)
+        elsif frame_floor.is_ceiling
+          next unless [HPXML::LocationAtticVented, HPXML::LocationAtticUnvented].include?(frame_floor.exterior_adjacent_to)
+        end
+
+        cost_mult += frame_floor.area
+      end
+    elsif cost_mult_type == 'Floor Area, Lighting (ft^2)'
+      cost_mult += hpxml.building_construction.conditioned_floor_area
+      hpxml.slabs.each do |slab|
+        next unless [HPXML::LocationGarage].include?(slab.interior_adjacent_to)
+
+        cost_mult += slab.area
+      end
+    elsif cost_mult_type == 'Roof Area (ft^2)'
+      hpxml.roofs.each do |roof|
+        cost_mult += roof.area
+      end
+    elsif cost_mult_type == 'Window Area (ft^2)'
+      hpxml.windows.each do |window|
+        cost_mult += window.area
+      end
+    elsif cost_mult_type == 'Door Area (ft^2)'
+      hpxml.doors.each do |door|
+        cost_mult += door.area
+      end
+    elsif cost_mult_type == 'Duct Surface Area (ft^2)'
+      hpxml.hvac_distributions.each do |hvac_distribution|
+        hvac_distribution.ducts.each do |duct|
+          cost_mult += duct.duct_surface_area
+        end
+      end
+    elsif cost_mult_type == 'Size, Heating System (kBtu/h)'
+      hpxml.heating_systems.each do |heating_system|
+        next if heating_system.heating_capacity.nil? # FIXME
+
+        cost_mult += UnitConversions.convert(heating_system.heating_capacity, 'btu/hr', 'kbtu/hr')
+      end
+      hpxml.heat_pumps.each do |heat_pump|
+        next if heat_pump.heating_capacity.nil? # FIXME
+
+        cost_mult += UnitConversions.convert(heat_pump.heating_capacity, 'btu/hr', 'kbtu/hr')
+      end
+    elsif cost_mult_type == 'Size, Heating Supplemental System (kBtu/h)'
+      heating_systems = {}
+      hpxml.heating_systems.each do |heating_system|
+        next if heating_system.fraction_heat_load_served == 1.0
+
+        heating_systems[heating_system] = heating_system.fraction_heat_load_served
+      end
+      if heating_systems.size == 2
+        heating_systems = heating_systems.sort_by{|k, v| v}
+        cost_mult += heating_systems.values[0] # lowest frac
+      end
+    elsif cost_mult_type == 'Size, Cooling System (kBtu/h)'
```

Comment:
I assume you know this, but this code won't actually work for ResStock because ResStock uses autosized HVAC systems and the calculated capacities are not currently populated in the in.xml HPXML file. Your tests are working only because you're referencing HPXML files with hardsized systems.

Reply:
I know that. The FIXMEs were used to indicate that. Where can we get calculated system sizes? Can we not populate them in the in.xml HPXML file?

Reply:
That is the goal, but it's a bit complicated. I'll give some thought to this. I don't think this issue needs to hold up the PR, but we'll obviously need to address it eventually.